lib/extensions/ar_virtual.rb

```diff
@@ -386,6 +386,14 @@ def virtual_attribute_names
       end
     end
 
+    def attributes_builder
+      @attributes_builder ||= ::ActiveRecord::AttributeSet::Builder.new(attribute_types, primary_key) do |name|
+        unless columns_hash.key?(name) || virtual_attribute?(name)
```

Comment:
+1 to original code thought and even still I think it looks good 😄  If there were a negation in there it would be a whole different story.